uber/templates/emails/pending.html

```diff
@@ -2,29 +2,54 @@
 {% block title %}Automated Emails Pending Approval{% endblock %}
 {% block content %}
 
-<h2> Pending Emails Requiring Approval </h2>
+<h2>Email Categories</h2>
 
-<table class="list">
+{% if not last_run_result %}
+<p><font color="red">
+    Note: Incomplete information is available about the number of emails requiring approval. Either the email daemon has
+    not run yet (should be a max wait of 5 minutes), or, email sending is disabled.
+</font></p>
+{% endif %}
+    
+<table class="table table-striped" data-page-size="9999999">
+<thead>
 <tr class="header">
-    <td>Subject</td>
-    <td>Sender</td>
-    <td>Examples</td>
-    <td>Approve</td>
+    <th>Subject</th>
+    <th>Sender</th>
+    <th>Number Sent</th>
+    <th>Number Queued</th>
+    <th>Examples</th>
+    <th>Approved?</th>
+    <th>Date Condition</th>
 </tr>
-{% for email in pending %}
+</thead>
+<tbody>
+{% for entry in automated_emails %}
     <tr>
-        <td>{{ email.subject }}</td>
-        <td>{{ email.sender }}</td>
-        <td><a href="pending_examples?subject={{ email.subject|urlencode }}">See Examples</a></td>
+        <td>{{ entry.automated_email.subject }}</td>
+        <td>{{ entry.automated_email.sender }}</td>
+        <td>{{ entry.num_sent }}</td>
+        <td>{% if entry.last_run_results != None %}{{ entry.last_run_results.unsent_because_unapproved }}{% else %}_{% endif %}</td>
+        <td><a href="pending_examples?ident={{ entry.automated_email.ident|urlencode }}">See Examples</a></td>
         <td>
-            <form method="post" action="approve">
-            {% csrf_token %}
-            <input type="hidden" name="subject" value="{{ email.subject }}" />
-            <input type="submit" value="Approve" />
-            </form>
+            {% if not entry.approved %}
+                <form method="post" action="approve">
+                {% csrf_token %}
+                <input type="hidden" name="ident" value="{{ entry.automated_email.ident }}" />
+                <input type="submit" value="Approve" />
+                </form>
+            {% else %}
+                {% if entry.doesnt_need_approval %}
```

Comment:
This if block will never be true. You define "approved" as: `not automated_email.needs_approval or automated_email.ident in approved_idents` but "doesnt_need_approval" is `not automated_email.needs_approval`. By those definitions, if an email is not included in `approved`, it will also never be included in `doesnt_need_approval`. This is what I was trying to point out earlier.